R/staggregate.R

```diff
@@ -28,80 +40,111 @@ daily_aggregation <- function(data, overlay_weights, daily_agg, time_interval='1
   # Data.table of weights
   weights_dt <- data.table::as.data.table(overlay_weights)
 
-  # Check if raster overlay_weights span prime meridian
-  is_pm <- FALSE
-  for(i in length(weights_dt$x)){
-    if(weights_dt[i,x] < .5 | weights_dt[i,x] > 359.5){
-      is_pm = TRUE # Check each x value
-      break # If near 0 value found exit loop
-    }
+  # read in climate data and coerce if not a spat raster
+  if(inherits(data, "SpatRaster")){
+
+    clim_stack <- data
+
+  } else{
+
+    clim_stack <- terra::rast(data)
+
   }
 
-  ## read in climate data
-  clim_raster <- raster::stack(data)
 
-  ## shift into 0 to 360 if not already in that format
-  if(raster::extent(clim_raster)@xmin < 0 - raster::xres(clim_raster) / 2) {
+  # Determine whether the climate data is in climate (0 to 360) or standard (-180 to 180) coordinates
+  if(terra::ext(clim_stack)$xmax > 180 + terra::xres(clim_stack)){
+    coord_alignment <- "climate"
+  } else{
+    coord_alignment <- "standard"
+  }
 
-    clim_raster_xmin <- raster::extent(clim_raster)@xmin - raster::xres(clim_raster) / 2
-    clim_raster_xmax <- raster::extent(clim_raster)@xmax + raster::xres(clim_raster) / 2
-    clim_raster_ymin <- raster::extent(clim_raster)@ymin - raster::yres(clim_raster) / 2
-    clim_raster_ymax <- raster::extent(clim_raster)@ymax + raster::yres(clim_raster) / 2
+  # Now, determine if the polygons were likely in a different coordinate system
+  if(coord_alignment == "standard"){
+    polygons_split <- (
+      max(weights_dt[,x]) > 180 - terra::xres(clim_stack) & # has cell next to 180
+      min(weights_dt[,x]) < -180 + terra::xres(clim_stack) # has cell next to -180
+    )
+  }
 
-    clim_raster1 <- raster::crop(clim_raster, c(clim_raster_xmin, min(clim_raster_xmax, 0), clim_raster_ymin, clim_raster_ymax))
+  if(coord_alignment == "climate"){
+    polygons_split <- (
+      max(weights_dt[,x]) > 360 - terra::xres(clim_stack) & # has cell next to 360
+      min(weights_dt[,x]) < terra::xres(clim_stack) # has cell next to 0
+    )
+  }
 
-    clim_raster1 <- raster::shift(clim_raster1, dx = 360)
+  # Check for the final case in which the polygons just span the entire globe
+  if(length(unique(weights_dt[,x] >= terra::ncol(clim_stack)))){
```

Comment:
I think there is a misplaced parenthesis or two in this line

Reply:
Great catch! Changed to `if(length(unique(weights_dt[,x])) >= terra::ncol(clim_stack))`